wagtail/wagtailimages/models.py

```diff
@@ -107,8 +118,18 @@ def get_willow_image(self):
         # Open file if it is closed
         close_file = False
         try:
+            image_file = self.file
+
             if self.file.closed:
-                self.file.open('rb')
+                # Reopen the file
+                if self.is_stored_locally():
+                    self.file.open('rb')
+                else:
+                    # Some external storage backends don't allow reopening
```

Comment:
Perhaps it might be simpler to just get a fresh ``File`` object for local files as well